graveler/staging.go

```diff
@@ -64,6 +66,33 @@ func (p *stagingManager) Drop(ctx context.Context, st StagingToken) error {
 	return err
 }
 
+func (p *stagingManager) DropByPrefix(ctx context.Context, st StagingToken, prefix Key) error {
+	upperBound := make(Key, len(prefix))
+	useUpperBound := false
+	copy(upperBound, prefix)
+	for i := len(prefix) - 1; i >= 0; i-- {
+		if upperBound[i] == math.MaxUint8 {
+			upperBound = upperBound[:i]
+		} else {
+			useUpperBound = true
+			upperBound[i] += 1
+			break
+		}
+	}
```

Comment:
1. can we extract this one to a different func
2. can we lookup the upper bound index and based on the value, return nil, return position, or make a copy and increment. (using bytes.LastIndexByte for MaxUInt8)